src/zenml/orchestrators/context_utils.py

```diff
@@ -49,3 +51,62 @@ def add_stack_as_metadata_context(
     for k, v in stack_dict.items():
         c_property = context.properties[k]  # type:ignore[attr-defined]
         c_property.field_value.string_value = v
+
+
+def add_schedule_as_metadata_context(
+        schedule: Schedule,
+        context: "pipeline_pb2.ContextSpec",  # type: ignore[valid-type]
+) -> None:
+    """
+
+    Args:
+        schedule: an instance of a Zenml Schedule object
+        context: a context proto message within a pipeline node
+    """
+    context.type.name = (   # type:ignore[attr-defined]
+        MetadataContextTypes.SCHEDULE.value
+    )
+
+    # Setting the name of the context
+    name = str(hash(schedule.json(sort_keys=True)))
+    context.name.field_value.string_value = name  # type:ignore[attr-defined]
+
+    # Setting the properties of the context
+    for k, v in schedule.dict().items():
+        c_property = context.properties[k]  # type:ignore[attr-defined]
+        if isinstance(v, int):
+            c_property.field_value.int_value = v
+        elif isinstance(v, float):
+            c_property.field_value.double_value = v
+        elif isinstance(v, str):
+            c_property.field_value.string_value = v
+        else:
+            c_property.field_value.string_value = str(v)
+
+
+def add_pydantic_object_as_metadata_context(
+        obj: "BaseModel",
+        context: "pipeline_pb2.ContextSpec",  # type: ignore[valid-type]
+) -> None:
+    """
+
+    Args:
+        obj: an instance of a pydantic object
+        context: a context proto message within a pipeline node
+    """
+    context.type.name = obj.__repr_name__().lower()
+    # Setting the name of the context
+    name = str(hash(obj.json(sort_keys=True)))
+    context.name.field_value.string_value = name  # type:ignore[attr-defined]
+
+    # Setting the properties of the context
+    for k, v in obj.dict().items():
+        c_property = context.properties[k]  # type:ignore[attr-defined]
```

Comment:
More of a question than a request: For all the execution properties of a step, we only use the `string_value` of the property and always do a `json.dumps(value)` to serialize the values to a string. Do you see any pros/cons to both solutions and should we maybe try to unify them? (Probably also @bcdurak, he's probably the expert here)

Reply:
Not sure I fully understand the question. @bcdurak you got anything?

Reply:
I really think we should unify too, but something we can do later